common/src/View/MapFrame.cpp

```diff
@@ -130,8 +131,29 @@ namespace TrenchBroom {
             m_updateLocker->Start();
 #endif
         }
+        
+        static RenderView* FindChildRenderView(wxWindow *current) {
+            for (wxWindow *child : current->GetChildren()) {
+                RenderView *canvas = dynamic_cast<RenderView*>(child);
+                if (canvas)
+                    return canvas;
+                
+                canvas = FindChildRenderView(child);
+                if (canvas)
+                    return canvas;
+            }
+            return nullptr;
+        }
 
         MapFrame::~MapFrame() {
+            // Search for a RenderView (wxGLCanvas subclass) and make it current.
+            RenderView* canvas = FindChildRenderView(this);
+            if (canvas != nullptr && m_contextManager != nullptr) {
+                wxGLContext* mainContext = m_contextManager->mainContext();
+                if (mainContext != nullptr)
+                    mainContext->SetCurrent(*canvas);
+            }
+            
             // Makes IsBeingDeleted() return true
             SendDestroyEvent();
```

Comment:
Hasn't this been sent already? I would have thought `MapFrame`'s destructor is running as a result of receiving a destroy event for the frame.